packages/core/js-client/src/clientPeer/__test__/client.spec.ts

```diff
@@ -11,7 +11,7 @@ describe('FluenceClient usage test suite', () => {
         await withClient(RELAY, {}, async (peer) => {
             // arrange
 
-            const result = await new Promise<string[]>((resolve, reject) => {
+            const result = await new Promise<string[]>(async (resolve, reject) => {
```

Comment:
I believe, there is a way to make this look a lot less ugly and indented and avoid using async inside the promise. You can do something like this;
```js
let resolve;
let reject;

const resultPromise = new Promise((res, rej) => {
  resolve = res;
  reject = rej
});

// ...

registerHandlersHelper(peer, particle, {
                    load: {
                        relay: () => {
                            return peer.getRelayPeerId();
                        },
                    },
                    callback: {
                        callback: (args: any) => {
                            const [val] = args;
                            resolve(val);
                        },
                        error: (args: any) => {
                            const [error] = args;
                            reject(error)
                        },
                    },
                });
                
// ...

const result = await resultPromise
```

Also it's `Promise<string[]>` for some reason, but seems like it should be `Promise<string>` judging by the `expect` down below

Reply:
Well, there are almost all tests written in deep intended format. Making them all less intended on 2 spaces in promise body is not a purpose of this PR. Also i think that proposed code a bit more complicated because this not a standart way to use promises. Probably promise's body could be refactored to a separate helper function as promise constructor itself, but i suggest to leave it for another PR.

Reply:
BTW changed promise result type

Reply:
It's not the indentation, the main thing is using async inside the promise constructor function. This is antipattern, and in fact what you are doing is in fact not a standard way of using promises you can read about it more details about it on the internet, this can cause unexepected behaviour. There are different ways of avoiding this issue, e.g. you can have an async IIFE inside of the promise constructor function.

You are adding async in THIS PR. So it kinda means you have to address it in this PR, not in a separate one

Reply:
Yes, actually you're right. I will refactor promises

Reply:
Pls check again

Reply:
Good. Thanks